lib/puppet/provider/neutron_subnet/neutron.rb

```diff
@@ -82,11 +82,10 @@ def self.parse_allocation_pool(values)
   def self.parse_host_routes(values)
     host_routes = []
     return [] if values.empty? or values == '[]'
-    # Strip brackets from output, needed after neutronclient >= 6.1.0
-    values = values.gsub('[', '').gsub(']', '')
-    for value in Array(values)
-      host_route = JSON.parse(value.gsub(/\\"/,'"').gsub('u\'', '"')
-                              .gsub('\'','"'))
+    values = JSON.parse(values.gsub(/\\"/,'"').gsub('u\'', '"')
+                              .gsub('\'','"').gsub(/^{/, "[{").gsub(/}$/, "}]"))
```

Comment:
In case where multiple host routes are delivered that are not in an array, the last gsub could solve this. I'm going to assume that all objects consisting of multiple routes are delivered in an array and changed it accordingly.